Review and update the testing doc page (#227)
exercism/ballerina: docs/TESTS.md

# Exercise testing

Follow the steps below to develop a solution for an exercise and test it locally.

## Download the exercise

> Open a Command Prompt and execute the command below to download the exercise.

Reviewer: ```suggestion
Open a Command Prompt and execute the command below with the name of the preferred exercise to download the exercise.
```

> >**Tip:** Replace the value of the `--exercise` property with the name of your preferred exercise.

Author: Removed.

```sh

> exercism download --track ballerina --exercise hello-world

Reviewer: ```suggestion
exercism download --track ballerina --exercise <exercise name>
```

```

## Develop your solution

1. Execute the command below to navigate to the directory of the newly downloaded exercise.

     >**Tip:** The directory location in the command below depends on your OS and your configured Exercism workspace. You can find out your workspace location by executing the `exercism configure` command.

     ```sh
     cd ~/exercism/ballerina/hello-world/
     ```

2. Follow the provided instructions and develop a solution for the exercise.

## Test your solution

> Some Ballerina exercises will have a single test while others will have multiple tests. The tests can be found in the respective test file of the `/tests` directory inside the root directory of the exercise.

Reviewer: ```suggestion
Some Ballerina exercises will have a single test, while others will have multiple tests. The tests can be found in the respective test file of the `/tests` directory inside the root directory of the exercise.
```

### Test exercise with a single test

Execute the command below to run the test of the exercise.

By default, only the first test is enabled. This is intentional as it allows you to focus on just making that one test pass. 
     
```sh
bal test
```

### Test exercise with multiple tests

Follow the steps below to test an exercise with multiple tests.

> 1. Once the first test passes, enable the next test by removing the `enable: false` property shown in the example below from the `@test:Config` annotation of it.

Reviewer: `shown in the example below from the `@test:Config` annotation of it.` bit too wordy

Author: Changed to `Once the first test passes, enable the next test by removing the `enable: false` property from the `@test:Config` annotation of it.`

Reviewer: IMO, a simple sentence as follows is good enough. 
```Once the first test passes, enable the next test by removing the `enable: false` property of the `@test:Config` annotation. ```

Author: Updated.

> 1. Once the first test passes, enable the next test by commenting the `enable: false` property from the `@test:Config` annotation below of it as shown in the example below.

Reviewer 2: Is commenting out the line the correct way? Shouldn't we ask them to remove the line completely?

Author: Changed to removing.

     ```ballerina
     @test:Config {

>      enable: false

Reviewer 3: ```suggestion
       enable: false
```

     }
     function nextTestCase() {
     }
     ``` 

2.  Execute the command below to run the enabled test.

   ```sh
     bal test
   ```

> 3. Repeat the two steps above to enable the remaining tests one by one and run them.

Reviewer: Why do we need to ask the users to run tests one by one? IMO it is unnecessary.

Author: That is the approach we have followed when designing Exercism exercises. @malinthar - Can you verify?

Reviewer 3: Exercism encourages test driven development. That means enable one test case and get it passed by gradually improving your solution(Not the case for every exercise). We don't have to specifically say to enable the tests one by one. We can just say `Repeat the two steps above to enable the remaining tests.`

Author: Updated.

> You have solved the exercise when all tests have been enabled and your implementation makes them all pass.

Reviewer: You have successfully completed the exercise by enabling all tests and ensuring that your implementation passes each one.
```suggestion
You have successfully completed the exercise by enabling all tests and making sure that your implementation passes each one.
```
